realm-annotations-processor/src/main/java/io/realm/processor/Utils.java

```diff
@@ -73,7 +76,52 @@ public static boolean isString(VariableElement field) {
     }
 
     /**
-     * Returns true if a given field type string is "java.lang.String", false otherwise.
+     * Returns {@code true} if a field is a primitive type, {@code false} otherwise.
+     */
+    public static boolean isPrimitiveType(String typeString) {
+        if (typeString == null) {
+            return false;
+        }
+        return typeString.equals("byte") || typeString.equals("short") || typeString.equals("int") ||
+                typeString.equals("long") || typeString.equals("float") || typeString.equals("double") ||
+                typeString.equals("boolean") || typeString.equals("char");
+    }
+
+    /**
+     * Returns {@code true} if a field is a boxed type, {@code false} otherwise.
+     */
+    public static boolean isBoxedType(String typeString) {
+        if (typeString == null) {
+            return false;
+        }
+        return typeString.equals(Byte.class.getName()) || typeString.equals(Short.class.getName()) ||
+                typeString.equals(Integer.class.getName()) || typeString.equals(Long.class.getName()) ||
+                typeString.equals(Float.class.getName()) || typeString.equals(Double.class.getName()) ||
+                typeString.equals(Boolean.class.getName());
+    }
+
+    /**
+     * Returns {@code true} if a field is a type of primitive types, {@code false} otherwise.
+     */
+    public static boolean isPrimitiveType(VariableElement field) {
+        if (field == null) {
```

Comment:
`VariableElement ` should never be null, we should not continue the annotation processor, throw IAE